R/boxr_auth.R

```diff
@@ -298,3 +298,95 @@ box_auth_on_attach <- function(auth_on_attach = FALSE) {
   }
 }
 
+#' Authenicate a box.com account with a JWT (JSON Web Token)
+#' 
+#' More doc-ing needed on setting up a JWT app, very similar to O-Auth2.
+#' @param config_file Path to JSON config file.
+#' @param user_id User ID number for Box account authorization.
+#' @importFrom jsonlite fromJSON
+#' @importFrom openssl read_key base64_encode
+#' @importFrom jose jwt_claim jwt_encode_sig
+#' @export
+box_auth_jwt <- function(config_file = "", user_id = "") {
+  if (config_file == "") {
```

Comment:
Hi Nate - I know I'm being picky here: `==` can be vectorized, would you consider:

```r
if (identical(config_file, "")) {
```

As this can return only a single `TRUE` or `FALSE`.

Reply:
Absolutely, I was mirroring the existing patterns in `box_auth`, but I think your suggestion is better. I'll update the existing places where `==` is used to check arguments in the auth functions. If left to my own devices I usually would use `NULL` over the empty string, do you have insight about one pattern being superior to the other?

Reply:
I think we are on the same page here - I like the `NULL` pattern paired with documenting the default, then using `rlang::%||%` (which we should consider importing internally, if we do not already). 

It's a philosophical question about how/when to change the API, but I would have no problem starting now with new functionality.